### PR TITLE
Change references to master/slave/black/white to more appropriate terms (SCP-3345)

### PR DESCRIPTION
**Pull request by bistline**
In line with the industry-wide move away from computing terms that have a racial context, this update renames methods/variables that contain references to the above terms with one of the following replacements:

* `master`: `main`, or `parent`
* `slave`: `worker`, or `child`
* `black` (used to denote denial/restriction, rather than the actual color): `deny`
* `white` (used to denote allowed/accepted, rather than the actual color): `allow`, or `permit`

There is an associated ticket (SCP-3346) to rename the `master` branch in all SCP Github repositories to `main`.  This work impacts all of our automation for CI/deployments, so is not included in this update.

This PR satisfies SCP-3345.

### CODE REVIEW

**Review by devonbush — APPROVED** [review]
Nice!  thanks for making the effort on this.  I'm approving, although I'd recommend we don't merge this until after the release tomorrow, since it touches a bunch of files and is non-urgent.  (That said, it's the release engineer's call

**Review by eweitz — APPROVED** [review]

**Comment by bistline** [comment]
> Nice! thanks for making the effort on this. I'm approving, although I'd recommend we don't merge this until after the release tomorrow, since it touches a bunch of files and is non-urgent. (That said, it's the release engineer's call

I've already decided that I want to hold this out of tomorrow's release.  I expect there to be regressions given the amount of refactoring that happened, so I want to see what build says.  Even if it's green, I still want to poke around a little more.

**Pull request merged**
